### PR TITLE
[vernac] Rename Vernacinterp to Vernacextend and move extension functions there

### PR DESCRIPTION
**Pull request by ejgallego**
This PR fixes an issue that was bugging me for some time, namely that
`Vernacinterp` really means `Vernacextend`.

We thus rename the file and move the associated functions there, which
were incorrectly placed in `Vernacentries`.

Note the beneficial effects on reducing the `.mli` API.

### CODE REVIEW

**Comment by ejgallego** [comment]
@ppedrot what is the proper way to fix the `top_printers` ?

**Comment by ppedrot** [comment]
What's the matter?

**Comment by ejgallego** [comment]
> What's the matter?

Two printers there should be moved from vinterp_add to vernac_extend.

**Comment by ejgallego** [comment]
See link https://github.com/coq/coq/blob/011d915ad22e5a97ab2177fbe1301655c2272c96/dev/top_printers.ml#L518

File: dev/top_printers.ml [hunk 1]
```diff
@@ -515,7 +515,7 @@ open Egramml
 
 let _ =
   try
-    Vernacinterp.vinterp_add false ("PrintConstr", 0)
+    Vernacextend.vinterp_add false ("PrintConstr", 0)
```
Comment (SkySkimmer) [inline dev/top_printers.ml]: TBH I don't know what this stuff is for, maybe we could just remove it and nobody would notice.

**Review by gares — APPROVED** [review]
stm change ok

File: dev/top_printers.ml (hunk 1 above)
Comment (ejgallego) [reply dev/top_printers.ml]: I guess it is used to Print a Constr from Coq? I'd guess @herbelin uses it.

Comment (ppedrot) [reply dev/top_printers.ml]: I would support that. I don't use them. @herbelin probably does though.

**Comment by ejgallego** [comment]
> @ejgallego Nah, they are a huge hack to add commands to the debugger. Not sure we actually want to do that.

I can remove them but I'd like to hear @herbelin blessing.

File: dev/top_printers.ml (hunk 1 above)
Comment (ppedrot) [reply dev/top_printers.ml]: (Nice race condition.)

**Comment by ejgallego** [comment]
I've ported the commands nevertheless, but they should be tested.

**Comment by ejgallego** [comment]
Seems ready.

**Review by ppedrot — APPROVED** [review]

**Comment by herbelin** [comment]
I did not use "PrintConstr" for years. OK to me.

**Comment by ejgallego** [comment]
> I did not use "PrintConstr" for years. OK to me.

It is still supported tho, if you think it should go away please submit a PR cleaning it up.

[Tho I like them as they are a good use case on the new API, maybe I'll add to the doc of `Vernacextend`]

**Comment by herbelin** [comment]
> It is still supported tho, if you think it should go away please submit a PR cleaning it up.

I don't think it need to go away.

Then I maybe misunderstood the other question:

> >    @ejgallego Nah, they are a huge hack to add commands to the debugger. Not sure we actually want to do that.

> I can remove them but I'd like to hear @herbelin blessing.

What is the question to me?

**Comment by herbelin** [comment]
You can assign it to me if you want.

A question though: there is already a file `vernacextend.mlp` (not linked to Coq though). That may create confusions, at least for human beings.

**Comment by ejgallego** [comment]
> A question though: there is already a file `vernacextend.mlp` (not linked to Coq though). That may create confusions, at least for human beings.

Indeed, even if the belong to totally different "compilation" domains; I guess that's OK as we are gonna move `grammar` to its own package for 8.10, if possible to preserve compat.

**Comment by ppedrot** [comment]
Indeed grammar is likely to be extruded in 8.10, so this shouldn't really matter. I'll merge.

**Comment by ppedrot** [comment]
@ejgallego This PR didn't run with the last CI, just in case can you rebase?

**Comment by ejgallego** [comment]
> @ejgallego This PR didn't run with the last CI, just in case can you rebase?

Done.

**Comment by ppedrot** [comment]
OK, I was just worried by paramcoq, and it passed the CI. Other failures are spurious, I'll merge.

**Pull request merged**